components/common/src/ui.rs

```diff
@@ -175,155 +374,48 @@ impl UI {
             false,
         )
     }
+}
 
-    pub fn begin<T: ToString>(&mut self, message: T) -> Result<()> {
-        Self::write_heading(&mut self.shell.out, Colour::Yellow, '»', message)
+impl Default for UI {
+    fn default() -> Self {
+        UI::default_with(Coloring::Auto, None)
     }
+}
 
-    pub fn end<T: ToString>(&mut self, message: T) -> Result<()> {
-        Self::write_heading(&mut self.shell.out, Colour::Blue, '★', message)
-    }
+impl UIWriter for UI {
+    type ProgressBar = ConsoleProgressBar;
 
-    pub fn is_a_tty(&self) -> bool {
-        self.shell.input.isatty && self.shell.out.isatty && self.shell.err.isatty
+    fn out(&mut self) -> &mut io::Write {
+        &mut self.shell.out as &mut io::Write
```

Comment:
Are these casts superfluous? It seems to build for me with just
```rust
        &mut self.shell.out
```

Reply:
I'm not entirely sure where the code was at when I had added them but at that time I did need them. Not anymore though, removed!